calicoctl/calicoctl/commands/ipam/split.go

```diff
@@ -0,0 +1,208 @@
+	if err != nil {
+		return err
+	}
+
+	// Check that the datastore is locked
+	ctx := context.Background()
+	locked, err := common.CheckLocked(ctx, client)
+	if err != nil {
+		return fmt.Errorf("Error while checking if datastore was locked: %s", err)
+	} else if !locked {
+		return fmt.Errorf("Datastore is not locked. Run the `calicoctl datastore migrate lock` command in order split the IP pools.")
+	}
+
+	// Find the specified IP pool.
+	poolList, err := client.IPPools().List(ctx, options.ListOptions{})
+	if err != nil {
+		return fmt.Errorf("Unable to list IP pools to find the pool specified by %s", parsedArgs["<CIDR>"])
+	}
+
+	var oldPool *apiv3.IPPool
+	for _, pool := range poolList.Items {
+		if pool.Spec.CIDR == parsedArgs["<CIDR>"] {
+			oldPool = &pool
+		}
+	}
+
+	if oldPool == nil {
+		return fmt.Errorf("Unable to find IP pool covering the specified CIDR %s", parsedArgs["<CIDR>"])
+	}
+
+	// Disable the specified IP pool.
+	oldPool.Spec.Disabled = true
+	oldPool, err = client.IPPools().Update(ctx, oldPool, options.SetOptions{})
+	if err != nil {
+		return fmt.Errorf("Error disabling IP pool %s", parsedArgs["<CIDR>"])
+	}
+
+	// Calculate the split pool CIDRs.
+	numString := parsedArgs["<NUMBER>"].(string)
+	splitNum, err := strconv.Atoi(numString)
+	if err != nil {
+		return fmt.Errorf("Error reading number to split IP pools into. %s is not a valid number: %v", numString, err)
+	}
+
+	splitCIDRs, err := splitCIDR(oldPool.Spec.CIDR, splitNum)
+	if err != nil {
+		return fmt.Errorf("Error splitting the CIDR %s into %d CIDRs: %v", oldPool.Spec.CIDR, splitNum, err)
+	}
+
+	// Create the new split pools using UnsafeCreate.
+	for i, cidr := range splitCIDRs {
+		splitPool := &apiv3.IPPool{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: fmt.Sprintf("split-%s-%d", oldPool.GetObjectMeta().GetName(), i+1),
+			},
+			Spec: apiv3.IPPoolSpec{
+				CIDR: cidr,
+				// All other fields are preserved.
+				BlockSize:        oldPool.Spec.BlockSize,
+				VXLANMode:        oldPool.Spec.VXLANMode,
+				IPIPMode:         oldPool.Spec.IPIPMode,
+				NATOutgoing:      oldPool.Spec.NATOutgoing,
+				DisableBGPExport: oldPool.Spec.DisableBGPExport,
+				NodeSelector:     oldPool.Spec.NodeSelector,
+				AllowedUses:      oldPool.Spec.AllowedUses,
+			},
+		}
+
+		_, err = client.IPPools().UnsafeCreate(ctx, splitPool, options.SetOptions{})
+		if err != nil {
+			return fmt.Errorf("Error using unsafe create to make split pool with cidr %s: %v", cidr, err)
+		}
+	}
+
+	// Delete the old IP pool.
+	_, err = client.IPPools().Delete(ctx, oldPool.GetObjectMeta().GetName(), options.DeleteOptions{})
+	if err != nil {
+		return fmt.Errorf("Error removing the IP pool that was split %s: %v", oldPool.GetObjectMeta().GetName(), err)
+	}
+
+	// Output follow-up directions.
+	fmt.Printf("IP Pool %s was successfully split into %d smaller pools. Please refer to the documentation for final steps.", oldPool.GetObjectMeta().GetName(), splitNum)
```

Comment:
We might want to print out more details on the pools that were created - e.g., their names.

I'm imagining some output like this:

```
IP Pool %s was successfully split into %d smaller pools.
Created default-0 with CIDR 192.168.0.0/17
Created default-1 with CIDR 192.168.128.0/17
```

Ideally, we'd also provide a `-o json` or `-o yaml` mode that simply printed out a json of the newly created pools so that this command could be more easily scripted against.